log4j-appender/src/test/java/org/apache/kafka/log4jappender/KafkaLog4jAppenderTest.java

```diff
@@ -66,22 +66,47 @@ public void testKafkaLog4jConfigs() {
 
 
     @Test
-    public void testLog4jAppends() throws UnsupportedEncodingException {
-        PropertyConfigurator.configure(getLog4jConfig());
+    public void testLog4jAppends() {
+        PropertyConfigurator.configure(getLog4jConfig(false));
 
         for (int i = 1; i <= 5; ++i) {
             logger.error(getMessage(i));
         }
 
         Assert.assertEquals(
-                5, ((MockKafkaLog4jAppender) (logger.getRootLogger().getAppender("KAFKA"))).getHistory().size());
+                5, ((MockKafkaLog4jAppender) (Logger.getRootLogger().getAppender("KAFKA"))).getHistory().size());
     }
 
-    private byte[] getMessage(int i) throws UnsupportedEncodingException {
-        return ("test_" + i).getBytes("UTF-8");
+    @Test(expected = RuntimeException.class)
+    public void testLog4jAppendsWithSyncSendAndSimulateProducerFailShouldThrowException() {
+        Properties props = getLog4jConfig(true);
+        props.put("log4j.appender.KAFKA.IgnoreExceptions", "false");
+        // property for mock appender
+        props.put("log4j.appender.KAFKA.FailSync", "true");
+        PropertyConfigurator.configure(props);
+
+        logger.error(getMessage(0));
+
+        Assert.fail();
```

Comment:
If logger.error call does not result in an exception, the test will fail anyway because of expected RuntimeException. If an exception is thrown, this line is not executed

Reply:
true, removing line